RequestService/RequestService.Handlers/GetAllJobsByFilterHandler.cs

```diff
@@ -0,0 +1,204 @@
+
+        public GetAllJobsByFilterHandler(
+            IRepository repository,
+            IAddressService addressService,
+            IJobFilteringService jobFilteringService,
+            IGroupService groupService)
+        {
+            _repository = repository;
+            _addressService = addressService;
+            _jobFilteringService = jobFilteringService;
+            _groupService = groupService;
+            _jobBasicDedupe_EqualityComparer = new JobBasicDedupe_EqualityComparer();
+        }
+
+        private async Task AddPostCodeForLocations(List<JobDTO> allJobs)
+        {
+            //add postcode for locations if locations exist
+            List<Location?> distinctLocations = allJobs
+                .Where(x => x.Location != null)
+                .Select(x => x.Location)
+                .Distinct().ToList();
+            List<LocationDetails> locationDetails = new List<LocationDetails>();
+
+            if (distinctLocations.Count > 0)
+            {
+                GetLocationsRequest locationRequest = new GetLocationsRequest()
+                {
+                    LocationsRequests = new HelpMyStreet.Contracts.AddressService.Request.LocationsRequest()
+                    {
+                        Locations = distinctLocations.Select(x => x.Value).ToList()
+                    }
+                };
+                var details = await _addressService.GetLocations(locationRequest);
+
+                if (details != null)
+                {
+                    locationDetails = details.LocationDetails;
+                }
+            }
+
+            if (locationDetails.Count > 0)
+            {
+                foreach (JobDTO j in allJobs.Where(x => x.Location != null))
+                {
+                    var location = locationDetails.First(x => x.Location == j.Location);
+
+                    if (location != null)
+                    {
+                        j.PostCode = location.Address.Postcode;
+                    }
+                }
+            }
+        }
+
+        public async Task<GetAllJobsByFilterResponse> Handle(GetAllJobsByFilterRequest request, CancellationToken cancellationToken)
+        {
+            if (!string.IsNullOrEmpty(request.Postcode))
+            {
+                request.Postcode = HelpMyStreet.Utils.Utils.PostcodeFormatter.FormatPostcode(request.Postcode);
+
+                try
+                {
+                    var postcodeValid = await _addressService.IsValidPostcode(request.Postcode, cancellationToken);
+                }
+                catch (HttpRequestException)
+                {
+                    throw new PostCodeException();
+                }
+            }
+
+            if (request.ExcludeSiblingsOfJobsAllocatedToUserID.HasValue)
+            {
+                if (request.JobStatuses?.JobStatuses.Count() == 0 || !request.JobStatuses.JobStatuses.Contains(JobStatuses.Open))
```

Comment:
If ExcludeSiblingsOfJobsAllocatedToUserID.HasValue, I think we want to allow only one JobStatus - which is JobStatuses.Open.  This code would currently allow JobStatuses.Open + JobStatuses.InProgress, which would be invalid when combined with a ExcludeSiblingsOfJobsAllocatedToUserID.